root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,2221 @@
+
+                                // Open-Redirect対策
+                                if (!string.IsNullOrEmpty(model.ReturnUrl)
+                                    && GetConfigParameter.GetConfigValue("OAuthAuthorizationServerEndpointsRootURI").IndexOf(model.ReturnUrl) != 1)
+                                {   
+                                    return RedirectToLocal(model.ReturnUrl); 
+                                }
+                                else
+                                {
+                                    return RedirectToAction("Index", "Home");
+                                }
+
+                            case SignInStatus.LockedOut:
+                                // ロックアウト
+                                return View("Lockout");
+
+                            case SignInStatus.RequiresVerification:
+                                // EmailConfirmedとは別の2FAが必要。
+
+                                // 検証を求める（2FAなど）。
+                                return this.RedirectToAction(
+                                    "SendCode", new
+                                    {
+                                        ReturnUrl = model.ReturnUrl,  // 戻り先のURL
+                                        RememberMe = model.RememberMe // アカウント記憶
+                                    });
+
+                            case SignInStatus.Failure:
+                            // サインイン失敗
+
+                            default:
+                                // その他
+                                // "無効なログイン試行です。"
+                                ModelState.AddModelError("", Resources.AccountController.Login);
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        // EmailConfirmed == false の場合、
+
+                        // メアド検証用のメールを送信して、
+                        this.SendConfirmEmail(user);
+
+                        // メッセージを設定
+                        ModelState.AddModelError("", Resources.AccountController.Login_emailconfirm);
+                    }
+                }
+            }
+            else
+            {
+                // AccountLoginViewModelの検証に失敗
+            }
+
+            // 再表示
+            return View(model);
+        }
+
+        #endregion
+
+        #region サインアウト
+
+        /// <summary>
+        /// サインアウト
+        /// Get: /Account/LogOff
+        /// </summary>
+        /// <returns>ActionResult(RedirectToAction)</returns>
+        [HttpGet]
+        [AllowAnonymous] // 空振りできるように。
+        public async Task<ActionResult> LogOff()
+        {
+            if (User.Identity.IsAuthenticated) // 空振りできるように。
+            {
+                // サインアウト（Cookieの削除）
+                AuthenticationManager.SignOut(DefaultAuthenticationTypes.ApplicationCookie);
+
+                // イベント・ログ出力
```

Comment:
@daisukenishino 

The developers might mix up the word called "イベント・ログ" and *Windows Event Log*.
So, it might be a good idea to change to the word called *トレースログ*.